app/controllers/tickets_controller.rb

```diff
@@ -33,7 +31,31 @@ def buy
     redirect_to root_path
   end
 
+  def edit
+    @events = Event.all
+  end
+
   def update
+    parameters = params.require(:ticket).permit(:place, :event_id, :price)
```

Comment:
Te walidacje pokazują nam, że w metodzie update dzieje się na tyle dużo, że warto wyciągnąć to do interaktora.
https://github.com/AaronLasseigne/active_interaction

Proszę potraktować to jako zadanie obowiązkowe.

metoda kontrolera powinna wyglądać mniej więcej tak:
```ruby
def update
  outcome = UpdateTicket.run(user: current_user, ticket: ticket, ticket_params: parameters)
  if outcome.valid?
    flash[:notice] = 'Your ticket has been edited.'
  else
    flash[:alert] = outcome.errors.full_messages.joins('. ')
  end
end
```

Reply:
A ten kod powinien być dodany do interaktora jako walidacje i usunięty z kontrolera:
```ruby
  def can_edit
    @ticket = Ticket.find(params[:id])
    if @ticket.owner_id != current_user.id
      flash[:alert] = 'You can edit only your own ticket.'
      redirect_to root_path
    end
    unless @ticket.bought_by.nil?
      flash[:alert] = 'Ticket has already been sold, so it can not be modified.'
      redirect_to root_path
    end
  end
```